pipelines/common/src/main/java/com/google/fhir/analytics/view/ViewSchema.java

```diff
@@ -160,4 +170,132 @@ public static void setValueInStatement(
+   * @param vDef The current ViewDefinition used to define the flat view
+   * @return List of populated GenericRecords
+   */
+  public static List<GenericRecord> setValueInRecord(RowList rows, ViewDefinition vDef) {
+    List<GenericRecord> recordList = Lists.newArrayList();
+
+    // Add each RowElement value to its respective column in the GenericRecord
+    Schema flatSchema = ViewSchema.getAvroSchema(vDef);
+    for (FlatRow flatRow : rows.getRows()) {
+      GenericRecord currentRecord = new GenericData.Record(flatSchema);
+      for (RowElement e : flatRow.getElements()) {
+        if (e.getPrimitive() != null) {
+          if (ViewApplicator.ID_TYPE.equals(e.getColumnInfo().getInferredType())) {
+            currentRecord.put(e.getColumnInfo().getName(), e.getString());
+          } else {
+            String elementType =
+                e.getColumnInfo().getType() == null ? "any" : e.getColumnInfo().getType();
+            switch (elementType) {
+              case "boolean":
+                currentRecord.put(e.getColumnInfo().getName(), e.<Boolean>getPrimitive());
+                break;
+              case "integer":
+              case "unsignedInt":
+              case "integer64":
+              case "decimal":
+                currentRecord.put(e.getColumnInfo().getName(), e.getPrimitive());
+                break;
+              case "date":
+              case "dateTime":
+              case "instant":
+              case "time":
+              case "base64Binary":
+              case "canonical":
+              case "code":
+              case "id":
+              case "markdown":
+              case "oid":
+              case "string":
+              case "uri":
+              case "url":
+              case "uuid":
+              default:
+                currentRecord.put(e.getColumnInfo().getName(), e.getString());
+                break;
+            }
+          }
+        } else {
+          // Currently arrays are not supported; also type inference is not enabled.
+          if (e.getColumnInfo().isCollection() || e.getColumnInfo().getType() instanceof String) {
+            if (e.getValues() == null || e.getValues().size() < 1) {
+              currentRecord.put(e.getColumnInfo().getName(), null);
+            } else {
+              String result = String.join(",", Arrays.toString(e.getValues().toArray()));
+              currentRecord.put(e.getColumnInfo().getName(), result);
+            }
+          } else {
+            // This happens when there is no value for a column with a primitive type.
+            currentRecord.put(e.getColumnInfo().getName(), null);
+          }
+        }
+      }
+      recordList.add(currentRecord);
+    }
+    return recordList;
+  }
+
+  /**
+   * Creates an Avro Schema for a given View Definition
+   *
+   * @param view the input view
+   * @return Avro Schema
+   */
+  public static Schema getAvroSchema(ViewDefinition view) {
```

Comment:
Please add a reference to `DefinitionToAvroVisitor` with a note that we should keep these two conversions consistent (I was thinking of some refactoring here too but that is not going to be very easy/clean, so for now we can keep it this way).

Reply:
Done.